crates/core/src/config.rs

```diff
@@ -107,15 +117,72 @@ impl Config {
 
     pub fn from_file(path: &str) -> Result<Self> {
         let content = fs::read_to_string(path)?;
+        let data_type =
+            match Path::new(path).extension().and_then(OsStr::to_str) {
+                Some(file_extension) => file_extension,
+                None => {
+                    return Err(eyre!(
+                        "No file extension found in provided config file"
+                    ))
+                }
+            };
+        Self::extract_from_file(data_type, content)
+    }
+
+    fn extract_from_file(data_type: &str, content: String) -> Result<Self> {
+        let config_data: Self = match data_type {
+            "toml" => toml::from_str(&content)?,
+            "json" => serde_json::from_str(&content)?,
+            _ => return Err(eyre!("Unsupported config file format")),
+        };
+
+        Ok(config_data)
+    }
 
-        if path.ends_with(".toml") {
-            let ret: Self = toml::from_str(&content)?;
-            Ok(ret)
-        } else if path.ends_with(".json") {
-            let ret: Self = serde_json::from_str(&content)?;
-            Ok(ret)
-        } else {
-            Err(eyre!("Unsupported config file format"))
+    pub async fn validate_params(&self) -> Result<()> {
+        self.validate()?;
+        Self::check_url(&self.eth_execution_rpc, Blockchain::Ethereum).await?;
+        Self::check_url(&self.starknet_rpc, Blockchain::Starknet).await?;
+        self.validate_data_dir()
+    }
+
+    async fn check_url(url: &str, blockchain: Blockchain) -> Result<()> {
```

Comment:
I'm not sure if sending HTTP requests is the right approach when we construct config objects. I think that we should only validate URLs in terms that they conform to the URLs spec. That way we check if the provided values are somewhat valid without introducing side effects and latency by sending a request, and beerus will fail to start anyway if we provide the wrong urls. What do you think?

Reply:
I wanted to have it as an additional layer check. If it is unnecessary, I can easily remove it. @sergey-melnychuk, please add your opinion

Reply:
Functional check is a good idea I think, but it can be done after the valid config is created.

It is actually a very good check - make sure that both Ethereum and Starknet endpoint use the same network. We might want to call `*_chainId` on both and see if the results are compatible.

Reply:
I will then add constants for each Mainnet/Sepolia Ethereum/Starknet Id and check if the provided URLs in Config get those Id's by calling *_chainId on them. If the returned Id's don't match, verification will fail. Based on that, *_blockNumber calls in check_url() are going to be replaced by *_chainId